src/Smalot/PdfParser/RawData/RawDataParser.php

```diff
@@ -723,15 +719,12 @@ protected function getRawObject(string $pdfData, int $offset = 0): array
                     // hexadecimal string object
                     $objtype = $char;
                     ++$offset;
-                    $pregResult = preg_match(
-                        '/^([0-9A-Fa-f\x09\x0a\x0c\x0d\x20]+)>/iU',
-                        substr($pdfData, $offset),
-                        $matches
-                    );
-                    if (('<' == $char) && 1 == $pregResult) {
+
+                    $span = strspn($pdfData, "0123456789abcdefABCDEF\x09\x0a\x0c\x0d\x20", $offset);
+                    if (('<' == $char) && $span > 0 && @$pdfData[$offset+$span] == '>') {
```

Comment:
Common case: unlimited substring (and a regexp) just to test, that string consists of 20 allowed chars followed by >.
That's right what strspn with $offset parameter do, and one small if do.